restaurant.js

```diff
@@ -0,0 +1,323 @@
+        .then(function (data) {
+        var temp = data;
+        console.log(temp.tableCardList, "type asks", typeof temp.tableCardList);
+        console.log(temp.tableCardList);
+        if (sessionStorage.getItem("menuCardList") == null) {
+            sessionStorage.setItem("menuCardList", JSON.stringify(temp.menuCardList));
+        }
+        if (sessionStorage.getItem("tableCardList") == null) {
+            sessionStorage.setItem("tableCardList", JSON.stringify(temp.tableCardList));
+        }
+        pageload();
+    });
+}
+getpagedata(pageload);
+function pageload() {
+    //storing for table 
+    var tableCardList = JSON.parse(sessionStorage.getItem("tableCardList"));
+    var tableCards = document.getElementById("Table-list");
+    tableCardList.forEach(function (tableCard) {
+        var nd = document.createElement('div');
+        nd.setAttribute("class", "table-card");
+        nd.setAttribute("id", "".concat(tableCard.id));
+        var tableCardHtml = "\n        <h2 class=\"table-card-title drop\" attr-key=".concat(tableCard.id, ">Table ").concat(tableCard.id + 1, "</h2>\n        <p class=\"table-card-para drop\" attr-key=").concat(tableCard.id, ">Total cost: <span id=").concat(tableCard.id.toString() + "c", " attr-key=").concat(tableCard.id, ">").concat(tableCard.tableTotal, "</span></p>\n        ");
+        nd.innerHTML = tableCardHtml;
+        if (tableCards !== null) {
+            tableCards.appendChild(nd);
+        }
+    });
+    var MenuCardList = JSON.parse(sessionStorage.getItem("menuCardList"));
+    var menucards = document.getElementById("Menu-list");
+    MenuCardList.forEach(function (menuCard) {
+        var nd = document.createElement('div');
+        nd.setAttribute("class", "menu-card");
+        nd.setAttribute("id", "".concat(menuCard.id));
+        nd.setAttribute("draggable", "true");
+        var menuCardHtml = "\n                <h2 attr-key=".concat(menuCard.id, ">").concat(menuCard.name, "</h2>\n                <p attr-key=").concat(menuCard.id, ">Total cost: <span id=\"total-cost\" attr-key=").concat(menuCard.id, ">").concat(menuCard.cost, "</span></p>\n                <p attr-key=").concat(menuCard.id, ">Item type: <span id=\"item-type\" attr-key=").concat(menuCard.id, ">").concat(menuCard.type, "</span></p>\n        ");
+        nd.innerHTML = menuCardHtml;
+        if (menucards !== null) {
+            menucards.appendChild(nd);
+        }
+    });
+    var ob = document.querySelectorAll(".table-card");
+    console.log("helloo", ob);
+    ob.forEach(function (obj) {
+        obj.addEventListener("click", function (e) {
+            var tableCardList = JSON.parse(sessionStorage.getItem("tableCardList"));
+            if (tableCardList[(parseInt(e.currentTarget.id))].tableTotal == 0 || false) {
+                document.getElementById("tableempty").style.display = "block";
+            }
+            else {
+                tablebodyreload((parseInt(e.currentTarget.id) + 1).toString());
```

Comment:
why are you adding 1 and passing it to tablebodyreload function and substracting 1 and using it in tablebodyreload function
why can't you just send the id directly